runtime/src/main/cpp/KString.cpp

```diff
@@ -752,8 +754,9 @@ OBJ_GETTER(Kotlin_String_plusImpl, KString thiz, KString other) {
   RuntimeAssert(other != nullptr, "other cannot be null");
   RuntimeAssert(thiz->type_info() == theStringTypeInfo, "Must be a string");
   RuntimeAssert(other->type_info() == theStringTypeInfo, "Must be a string");
-  KInt result_length = thiz->count_ + other->count_;
-  if (result_length < thiz->count_ || result_length < other->count_) {
+  uint32_t result_length = thiz->count_ + other->count_;
+  if (result_length < thiz->count_ || result_length < other->count_ ||
+      result_length > static_cast<uint32_t>(std::numeric_limits<int32_t>::max())) {
```

Comment:
Can we reduce the number of checks here?

Reply:
Good catch, thanks!